swift/app.py

```diff
@@ -38,3 +40,31 @@ def frames(self) -> Iterable[QWidget]:
             An iterable object of Frame objects for showing.
         """
         return ()
+
+    def broadcast(self, channelName: str, content: Any):
+        """Broadcasts the content to the target channel.
+
+        Args:
+            channelName: Target channel name.
+            content: Content to be broadcast.
+        
+        """
+        try:
+            content = json.dumps(content)
+        except TypeError as e:
+            print(f"swift.app.broadcast(): {e!r}")
+            return
+        self.broadcastRequested.emit(channelName, content)
+
+    def receivedSlot(self, channelName: str, content: Any):
+        """This will be overridden by child classes."""
+
+    @pyqtSlot(str, str)
+    def _receivedMessage(self, channelName: str, msg: str):
+        """This is connected to self.received signal."""
+        try:
+            msg = json.loads(msg)
+        except json.JSONDecodeError as e:
+            print(f"swift.app._receivedMessage(): {e!r}")
+            return
+        self.receivedSlot(channelName, msg)
```

Comment:
This is just a preference, but in this case you may use `else:` block because it only has a single line (line 70).
I prefer to use `else:` block unless the block is too big.

However, you can keep this code, no problem!

Reply:
Oh, I agree with you. I will change it.